### PR TITLE
Add KHR/KHX/NN extensions to threading layer

### PR DESCRIPTION
**Pull request by mark-lunarg**
These were skipped due to problems in codegen. Added support for new objects into the layer, fixed codegen.

### CODE REVIEW

File: scripts/threading_generator.py [hunk 1]
```diff
@@ -392,6 +398,7 @@ def genCmd(self, cmdinfo, name):
             'vkFreeCommandBuffers',
             'vkCreateDebugReportCallbackEXT',
             'vkDestroyDebugReportCallbackEXT',
+            'vkCreateSwapChainKHR',
```
Comment (MarkY-LunarG) [inline scripts/threading_generator.py]: We probably need a hook into vkCreateSharedSwapchainKHR as well.

Comment (stroyan) [inline scripts/threading_generator.py]: This adds a declaration for a function that would go into layers/threading.cpp.
But I don't see your changes to that file.

Comment (mark-lunarg) [inline scripts/threading_generator.py]: @MarkY-LunarG , CreateSharedSwapchains is covered by this layer.  @stroyan, I'll add the routine for CreateSwapChainKHR.

**Comment by mark-lunarg** [comment]
@stroyan, removed CreateSwapchain from the special_functions list.  Also removed the hard-coded object lists and just checked the handles -- I noticed that PhysicalDevice was purposefully skipped and kept things that way, but also noticed that VkDescriptorSet was missing from the list of non-dispatchable objects and added that in (along with supporting changes).  Things seem fine this way, is there some non-obvious reason this object should be excluded?

**Comment by stroyan** [comment]
The hard-coded lists of object types evolved as the list of externsync parameters expanded.
There is no need to use ReadObject counters for types that never appear in externsync parameters, (or for types that only appear as externsync parameters with WriteObject counters).  But it seems that all the handle types now are listed as externsync at least some of the time.  Dropping the hardcoded lists will make it a bit easier to maintain when new handle types appear.

**Pull request merged**